contracts/SuiBridge.sol

```diff
@@ -269,6 +269,8 @@ contract SuiBridge is
             vault.transferERC20(tokenAddress, targetAddress, amount);
         }
 
+        // Do we want to move the limit checks earlier, e.g. even for signature verification?
+        // in the event of exhaustion of transfer allowance, the user can pay fewer gas fees
```

Comment:
I wouldn't do this in the signature verification, but we could add a "limit check" modifier to the "transferTokensWithSignatures" function that would accomplish the same thing but keep the separation of responsibility a bit cleaner. If we added a "willUSDAmountExceedLimit" (like mentioned above) we would also need to add a new "willTokenAmountExceedLimit" so the USD logic stays in the Limiter contract.

Reply:
sounds good to me. Do you think the gas saving is worth this change?